src/libraries/Native/Unix/Common/pal_compiler.h

```diff
@@ -15,7 +15,7 @@
 #define c_static_assert_msg(e, msg) _Static_assert((e), msg)
 #define c_static_assert(e) c_static_assert_msg(e, "")
 #else
-#define c_static_assert_msg(e, msg) typedef char __c_static_assert__[(e)?1:-1]
+#define c_static_assert_msg(e, msg) typedef char __c_static_assert__[1]
```

Comment:
This is makes the `c_static_assert_msg` macro a no-op.

Reply:
Is it also a no-op in master branch? I was getting:

```sh
error: size of array '__c_static_assert__' is negative
```

most of the supported platforms compile with other branches of this #if-#elif block.

Reply:
This error means that the `static_assert` worked as expected. The invariant that it was checking was not holding anymore. I understand that the error is not exactly intuitive, but it is the best we can do without built-in `static_assert`.

You should fix the invariant that it was complaining about.

Reply:
The only failing assertion was:

```c
c_static_assert(PAL_LOG_CRON == LOG_CRON);
```

on Linux, it is `9<<3` (72) and SmartOS has `15<<3` (120). I have `#ifdef`'d it for `TARGET_SUNOS`.